nf_core/download.py

```diff
@@ -270,53 +417,128 @@ def prompt_use_singularity_cachedir(self):
-                            f"[blue bold]?[/] [bold]Add to [blue not bold]~/{os.path.basename(bashrc_path)}[/] ?[/]"
+                            f"[blue bold]?[/] [bold]Add to [blue not bold]~/{os.path.basename(shellprofile_path)}[/] ?[/]"
                         )
                         if append_to_file:
-                            with open(os.path.expanduser(bashrc_path), "a") as f:
+                            with open(os.path.expanduser(shellprofile_path), "a") as f:
                                 f.write(
                                     "\n\n#######################################\n"
                                     f"## Added by `nf-core download` v{nf_core.__version__} ##\n"
                                     + f'export NXF_SINGULARITY_CACHEDIR="{cachedir_path}"'
                                     + "\n#######################################\n"
                                 )
-                            log.info(f"Successfully wrote to [blue]{bashrc_path}[/]")
+                            log.info(f"Successfully wrote to [blue]{shellprofile_path}[/]")
                             log.warning(
                                 "You will need reload your terminal after the download completes for this to take effect."
                             )
 
-    def prompt_singularity_cachedir_only(self):
+    def prompt_singularity_cachedir_utilization(self):
         """Ask if we should *only* use $NXF_SINGULARITY_CACHEDIR without copying into target"""
         if (
-            self.singularity_cache_only is None
+            self.singularity_cache is None  # no choice regarding singularity cache has been made.
             and self.container == "singularity"
             and os.environ.get("NXF_SINGULARITY_CACHEDIR") is not None
+            and stderr.is_interactive
         ):
             stderr.print(
-                "\nIf you are working on the same system where you will run Nextflow, you can leave the downloaded images in the "
-                "[blue not bold]$NXF_SINGULARITY_CACHEDIR[/] folder, Nextflow will automatically find them. "
+                "\nIf you are working on the same system where you will run Nextflow, you can amend the downloaded images to the ones in the"
+                "[blue not bold]$NXF_SINGULARITY_CACHEDIR[/] folder, Nextflow will automatically find them."
                 "However if you will transfer the downloaded files to a different system then they should be copied to the target folder."
             )
-            self.singularity_cache_only = rich.prompt.Confirm.ask(
-                "[blue bold]?[/] [bold]Copy singularity images from [blue not bold]$NXF_SINGULARITY_CACHEDIR[/] to the target folder?[/]"
-            )
+            self.singularity_cache = questionary.select(
+                "[blue bold]?[/] [bold]Copy singularity images from [blue not bold]$NXF_SINGULARITY_CACHEDIR[/] to the target folder or amend new images to the cache?[/]",
```

Comment:
```suggestion
                "Copy singularity images from $NXF_SINGULARITY_CACHEDIR to the target folder or amend new images to the cache?",
```
this syntax doesn't work in questionary it looks like

Reply:
It does work in my terminal 🤔